misk-service/src/main/kotlin/misk/CoordinatedService.kt

```diff
@@ -32,6 +32,7 @@ internal class CoordinatedService(
 
             override fun failed(from: State, failure: Throwable) {
               outerService.notifyFailed(failure)
+              directDependsOn.forEach { it.stopIfReady() }
```

Comment:
Starting any downstream dependencies after a start failure definitely makes no sense, but I think it makes sense to stop as many upstream dependencies as possible after a stop failure. There are little benefits to hang on a stop failure and allow the upstream to run.